crates/c-api/include/wasmtime/profiling.h

```diff
@@ -0,0 +1,36 @@
+/**
+ * \file wasmtime/profiling.h
+ *
+ * \brief API for Wasmtime guest profiler
+ */
+
+#ifndef WASMTIME_PROFILING_H
+#define WASMTIME_PROFILING_H
+
+#include <wasm.h>
+#include <wasmtime/error.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct wasmtime_guestprofiler_t wasmtime_guestprofiler_t;
+
+WASM_API_EXTERN void wasmtime_guestprofiler_delete(
+    /* own */ wasmtime_guestprofiler_t *guestprofiler);
+WASM_API_EXTERN /* own */ wasmtime_guestprofiler_t *
+wasmtime_guestprofiler_new(wasm_name_t *module_name, uint64_t interval_nanos,
+                           size_t modules_size, wasm_name_t **modules_name,
+                           wasmtime_module_t **modules_module);
```

Comment:
Seems mixed, eg. wasm_byte_vec_new takes size first. I can change this but I feel that because this length applies to two following arguments it's clearer this way.

Reply:
On second though, it might be better to define `struct {wasm_name_t*, wasmtime_module_t*}` and pass pointer to that?

Reply:
I changed to passing tuple as struct instead of column-major pointers.